core/finality-grandpa/src/lib.rs

```diff
@@ -777,3 +750,17 @@ pub fn run_grandpa<B, E, Block: BlockT<Hash=H256>, N, RA, SC, X>(
 {
 	run_grandpa_voter(grandpa_params)
 }
+
+/// Checks if this node is a voter in the given voter set.
+///
+/// Returns the key pair of the node that is being used in the current voter set or `None`.
+fn is_voter(
```

Comment:
should return an iterator of all keys that we own and that are authorities, rather than just the first one. cc @folsen

Reply:
Filing as issue here from looking at it: https://github.com/paritytech/substrate/issues/3324